### PR TITLE
tidb: add a new entry in release notes 5.1.4

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Add a new entry in release notes 5.1.4

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/11127
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-5.1.4.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ TiDB version: 5.1.4
     - Fix the `INDEX OUT OF RANGE` error for a MPP query after deleting an empty `dual table` [#28250](https://github.com/pingcap/tidb/issues/28250)
     - Fix the issue of false positive error log `invalid cop task execution summaries length` for MPP queries [#1791](https://github.com/pingcap/tics/issues/1791)
     - Fix the issue that SET GLOBAL tidb_skip_isolation_level_check=1 doesn't affect new session settings [#27897](https://github.com/pingcap/tidb/issues/27897)
+    - Fix the `index out of range` issue that occurs when `tiup bench` runs a long time [#26832](https://github.com/pingcap/tidb/issues/26832)
```
Comment (TomShawn) [inline releases/release-5.1.4.md:82]:
```suggestion
    - Fix the `index out of range` issue that occurs when `tiup bench` runs for a long time [#26832](https://github.com/pingcap/tidb/issues/26832)
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**